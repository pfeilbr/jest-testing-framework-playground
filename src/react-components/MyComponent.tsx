import * as React from "react";
export default class MyComponent extends React.Component<
  { title: string },
  { count: number }
> {
  constructor(props) {
    super(props);
    this.state = { count: 0 };
  }

  handleClick(event) {
    this.setState(prevState => ({ count: prevState.count + 1 }));
  }

  render() {
    return (
      <div>
        <h1>{this.props.title}</h1>
        <div>
          <label htmlFor="count">Count</label>
          <input
            type="text"
            name="count"
            readOnly
            value={`${this.state.count}`}
          />
          <button onClick={this.handleClick.bind(this)}>click</button>
        </div>
      </div>
    );
  }
}
