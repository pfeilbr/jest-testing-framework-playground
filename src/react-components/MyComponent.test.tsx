import React from "react";
import MyComponent from "./MyComponent";
import * as renderer from "react-test-renderer";
import "jest-dom/extend-expect";
import { render, fireEvent } from "react-testing-library";

test("MyComponent works", async () => {
  const component = renderer.create(<MyComponent title="Hello World" />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  const { getByText, getByLabelText } = render(
    <MyComponent title="Hello World" />
  );
  expect(getByText(/hello/i)).toHaveTextContent("Hello World");

  fireEvent.click(getByText(/click/i));
  expect(document.querySelector("[name=count]")).toHaveAttribute("value", "1");
});
