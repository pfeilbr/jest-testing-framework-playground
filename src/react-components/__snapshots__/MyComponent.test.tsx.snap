// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div>
  <h1>
    Hello World
  </h1>
  <div>
    <label
      htmlFor="count"
    >
      Count
    </label>
    <input
      name="count"
      readOnly={true}
      type="text"
      value="0"
    />
    <button
      onClick={[Function]}
    >
      click
    </button>
  </div>
</div>
`;

exports[`MyComponent works 1`] = `
<div>
  <h1>
    Hello World
  </h1>
  <div>
    <label
      htmlFor="count"
    >
      Count
    </label>
    <input
      name="count"
      readOnly={true}
      type="text"
      value="0"
    />
    <button
      onClick={[Function]}
    >
      click
    </button>
  </div>
</div>
`;
